rfcs/proposed/task_group_dynamic_dependencies/README.md

```diff
@@ -0,0 +1,378 @@
+performs a parallel recursive decomposition of a range. Currently,
+the oneTBB algorithms, such as tbb::parallel_for, are implemented using the non-public,
+low-level tasking API, rather than `tbb::task_group`. This low-level tasking API
+puts the responsibility for dependence tracking and memory
+management of tasks on developers. While it allows the oneTBB development team to build highly optimized
+algorithms, a simpler set of interfaces can be provided for the 
+users. Recursive parallel algorithms are one of the primary cases that we want 
+our task_group extension to cover.
+
+The key capability required for recursive decomposition is the ability to 
+create work while executing a task and insert this newly created work before 
+the (perhaps already submitted) successors of the currently executing task. 
+A simple example is a merge sort. As shown in the figure
+below, the top-level algorithm breaks a collection into two pieces and 
+creates three tasks:
+
+1. A task to sort the left half.
+2. A task to sort the right half.
+3. A task to merge the halves once they are sorted. 
+
+In a recursive merge sort, each of the sort tasks recursively takes the same 
+approach to sort its portions of the collection. The top-level task (and 
+subsequent recursively generated tasks) must be able to create new tasks 
+and then update the graph for their outer merge task to wait for the 
+newly created subtasks to complete.
+
+<img src="merge_sort.png" width=800>
+
+A key point of this recursive parallel algorithm is the requirement to change 
+the predecessors of the merge tasks. However, the merge tasks are already 
+submitted when their predecessors are modified. As mentioned in the previous 
+section, updating the predecessors of a submitted task can be
+risky due to the potential for a race condition. However, in this case,
+it is safe to add or change predecessors to the merge task. 
+This is because the merge task cannot start execution until all of its current predecessors 
+complete. Those predecessors are the tasks responsible for modifying the merge task dependencies. 
+
+Therefore, we propose a limited extension that allows transferring 
+all the successors of the currently executing task to become the successors 
+of a different created task. This function can only access the successors
+of the currently executing task, and those tasks are prevented from executing 
+by a dependence on the current task itself, so we can ensure that we can safely 
+update the incoming dependencies for those tasks without worrying about
+potential race conditions. 
+
+One possible spelling for this function would be `transfer_successors_to(h)`, 
+Where `h` is a `task_handle` to a created task, and the 
+`transfer_successors_to` function must be called from within a task. Calling
+this function from outside a task or passing anything other than a `task_handle`
+representing a task in the created state is undefined behavior.
```

Comment:
I think we need to add information about possible limitations (or lack of limitations) for the task states represented by `task_handle` to which we are transferring the successors. Does it sufficient to allow only the handles representing created tasks, or other 3 states are also allowed (i.e. transferring successors to the task_handle that was already submitted to `task_group::run`).